### ISSUE TITLE
[JS] experimental API `pull` with `max_bytes` returns only first message

### ISSUE THREAD

**Issue by aricart** [nats-io/nats-server#3816]
## Defect

Given a consumer that has 100 pending messages: 

```javascript
{
  type: "io.nats.jetstream.api.v1.consumer_info_response",
  stream_name: "K54T0380942R0POQHUKQ2I",
  name: "K54T0380942R0POQHUKQ2I",
  created: "2023-01-25T20:54:11.822784Z",
  config: {
    durable_name: "K54T0380942R0POQHUKQ2I",
    name: "K54T0380942R0POQHUKQ2I",
    deliver_policy: "all",
    ack_policy: "none",
    max_deliver: -1,
    replay_policy: "instant",
    max_waiting: 512,
    num_replicas: 0
  },
  delivered: { consumer_seq: 0, stream_seq: 0 },
  ack_floor: { consumer_seq: 0, stream_seq: 0 },
  num_ack_pending: 0,
  num_redelivered: 0,
  num_waiting: 0,
  num_pending: 100
}
```

Doing a pull request like this:

```bash
PUB $JS.API.CONSUMER.MSG.NEXT.K54T0380942R0POQHUKQ2I.K54T0380942R0POQHUKQ2I _INBOX.K54T0380942R0POQHUL57X 78␍␊{"batch":0,"max_bytes":1024,"idle_heartbeat":5000000000,"expires":10000000000}␍␊
```

Will yield a single message from the server. The server will not finalize the request, and no heartbeats are sent. The client will fail because of missed heartbeats. If `batch` is set to something, ie `1000` - the request will succeed, heartbeat properly and send the finalization message telling how many bytes/messages are pending before the expiration.

This happens all the way from 2.9.4  including current dev (2.11.x)

#### Steps or code to reproduce the issue:

Sample program to exercise this: 
https://github.com/aricart/js-simplification-examples/blob/main/max_bytes_issue.ts

```bash
nats-server -js -sd /tmp/jsdata &
deno run -A https://raw.githubusercontent.com/aricart/js-simplification-examples/main/max_bytes_issue.ts
{
  type: "io.nats.jetstream.api.v1.consumer_info_response",
  stream_name: "K54T0380942R0POQHUKQ2I",
  name: "K54T0380942R0POQHUKQ2I",
  created: "2023-01-25T20:54:11.822784Z",
  config: {
    durable_name: "K54T0380942R0POQHUKQ2I",
    name: "K54T0380942R0POQHUKQ2I",
    deliver_policy: "all",
    ack_policy: "none",
    max_deliver: -1,
    replay_policy: "instant",
    max_waiting: 512,
    num_replicas: 0
  },
  delivered: { consumer_seq: 0, stream_seq: 0 },
  ack_floor: { consumer_seq: 0, stream_seq: 0 },
  num_ack_pending: 0,
  num_redelivered: 0,
  num_waiting: 0,
  num_pending: 100
}
< SUB _INBOX.K54T0380942R0POQHUL57X 2␍␊PUB $JS.API.CONSUMER.MSG.NEXT.K54T0380942R0POQHUKQ2I.K54T0380942R0POQHUKQ2I _INBOX.K54T0380942R0POQHUL57X 78␍␊{"batch":0,"max_bytes":1024,"idle_heartbeat":5000000000,"expires":10000000000}␍␊
> MSG K54T0380942R0POQHUKQ2I.x 2 $JS.ACK.K54T0380942R0POQHUKQ2I.K54T0380942R0POQHUKQ2I.1.1.1.1674680051829863000.99 10␍␊␍␊
1
> PING␍␊
< PONG␍␊
```

#### Expected result:
The 100 messages should have been delivered and the protocol messages signaling a heartbeat and completion of the request should have been returned to the client.

**Comment by derekcollison** [comment]
@Jarema possible to take a look?

**Comment by Jarema** [comment]
yes. Will take look.

**Comment by aricart** [comment]
One consideration before we go in a loop on this is that while the idea of having a request that only has `max_bytes` sounds right, is possibly wrong. Let me explain.

- Processing a message takes N amount of time
- max wait for ack is set to some value
- maximum number of messages that can be processed by a client are clamped by those to values when acks are required

If that is the case, requests may not require the number of messages but they may specify them anyway. This means that the API should allow specifying `batch` and `max_bytes` because it will be required anyway.

**Comment by Jarema** [comment]
We can do it in two steps:
1. If max_bytes is specified, add sentiel that informs user how many bytes were pending. Then we have the same behaviour for `max_bytes` and `batch` not fulfilled. That would be useful to clients no matter what. And would not affect traffic if only `batch` is set.
2. Consider disabling setting batch to 1 by default if `max_bytes` is passed.

**Comment by derekcollison** [comment]
Do we want/need these for a 2.9.12 release?

**Comment by wallyqs** [comment]
I think we should revise this within the scope of v2.10 and not as the patch release from next week.

**Comment by Jarema** [comment]
Yes, that is definitely not for the patch.
It can break some user simplification algorithms.

#3822

**Comment by wallyqs** [comment]
Fixed via #3822
